Frends.Zip.ExtractArchive/Frends.Zip.ExtractArchive/ExtractArchive.cs

```diff
@@ -50,16 +50,23 @@ public static UnzipOutput ExtractArchive([PropertyTab] UnzipInputProperties inpu
                         if (File.Exists(Path.Combine(input.DestinationDirectory, z.FileName)))
                         {
                             // Find a filename that does not exist. 
-                            var FullPath = Extensions.GetNewFilename(Path.Combine(Path.GetDirectoryName(input.DestinationDirectory), z.FileName), z.FileName, cancellationToken);
-                            path = FullPath;
+                            path = Extensions.GetNewFilename(Path.Combine(input.DestinationDirectory, z.FileName), cancellationToken);
 
-                            using var fs = new FileStream(FullPath, FileMode.Create, FileAccess.Write); z.Extract(fs);
+                            using var fs = new FileStream(path, FileMode.Create, FileAccess.Write);
+                            z.Extract(fs);
+                        }
+                        else
+                        {
+                            z.Extract(input.DestinationDirectory);
                         }
-                        else z.Extract(input.DestinationDirectory);
                     }
                     break;
             }
         }
+
+        if (options.DeleteZipFileAfterExtract)
+            File.Delete(input.SourceFile);
```

Comment:
Could there be an exception handler that lets the user know the extraction was successful but the deletion failed due to missing permissions, for example.